### PR TITLE
add new lr_scheduler_type: step_lr

### PR DESCRIPTION
**Pull request by Xingfu-Yi**
# What does this PR do?

Added a new learning rate scheduler: `step_lr`. This was implemented through the `TYPE_TO_SCHEDULER_FUNCTION get_step_lr`, based on the existing class `torch.optim.lr_scheduler.StepLR`. It retains the same parameters as other functions and uses `num_training_steps` to represent `step_size`, ensuring consistency.

### CODE REVIEW

**Comment by amyeroberts** [comment]
Hi @Xingfu-Yi, thanks for opening a PR! 

Could you provide some more context around this addition - ideally links to related feature requests, paper and research? 

cc @younesbelkada @muellerzr
